src/main/java/seedu/address/logic/commands/internship/role/EditInternshipRoleCommand.java

```diff
@@ -60,9 +73,25 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_INVALID_ROLE);
         }
 
-        InternshipRole roleWithNewOutcome = roleToEdit.getNewInternshipRoleWithOutcome(newOutcome);
-        model.setInternshipRole(roleToEdit, roleWithNewOutcome);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(roleWithNewOutcome)));
+        InternshipRole editedRole = roleToEdit;
+        if (newCycle != null) {
+            editedRole = editedRole.getNewInternshipRoleWithCycle(newCycle);
+        }
+        if (!newDescription.equals("")) {
```

Comment:
should just ret null from parser if the rest are returning null, a bit odd to have only one using .equals("")